### PR TITLE
Fix download of artifacts

### PR DESCRIPTION
**Pull request by daniel-de-vera**
This PR implements the changes described in https://github.com/signadot/cli/pull/124#discussion_r1633829097.
It is stacked on:

- https://github.com/signadot/go-sdk/pull/49 (required changes in the SDK)
- https://github.com/signadot/cli/pull/126 (minor changes in job logs)

E.g.:
```bash
signadot artifact download --job hotrod-cypress-e2e-ivjkk3y hotrod/cypress/videos/hotrod.cy.js.mp4 -o /tmp/hotrod.cy.js.mp4
File saved successfully at /tmp/hotrod.cy.js.mp4

signadot artifact download --job hotrod-cypress-e2e-ivjkk3y @stdout -o /tmp/stdout
File saved successfully at /tmp/stdout
```

### CODE REVIEW

File: internal/config/api.go [hunk 1]
```diff
@@ -116,21 +115,15 @@ func (a *API) InitAPITransport() error {
 	return nil
 }
 
-func (a *API) APIClientWithCustomTransport(conf *transport.APIConfig, execute func(client *client.SignadotAPI) error) error {
+func (a *API) APIClientWithCustomTransport(conf *transport.APIConfig,
+	execute func(client *client.SignadotAPI) error) error {
 	if err := a.init(); err != nil {
-		return nil
+		return err
```
Comment (jmsktm) [inline internal/config/api.go:121]: good catch

File: internal/command/artifact/download.go [hunk 2]
```diff
@@ -53,31 +53,31 @@ func download(cfg *config.ArtifactDownload, out io.Writer, artifactPath string)
 
 	params := artifacts.
 		NewDownloadJobAttemptArtifactParams().
+		WithContext(ctx).
+		WithTimeout(2 * time.Minute).
```
Comment (jmsktm) [inline internal/command/artifact/download.go:57]: 2 min seems reasonable 👍. Can tune as needed.

**Review by jmsktm — APPROVED** [review]
LGTM

File: internal/command/artifact/download.go (hunk 2 above)
Comment (daniel-de-vera) [reply internal/command/artifact/download.go:57]: Yes, it's worth noting that the default timeout in the SDK is 30 secs (I didn't know this).

Comment (davixcky) [reply internal/command/artifact/download.go:57]: I think for large compressed files? Or videos?

Comment (daniel-de-vera) [reply internal/command/artifact/download.go:57]: I mean, it's hard to know what's the right value.. I can increase it a little bit.

Comment (daniel-de-vera) [reply internal/command/artifact/download.go]: It's done here https://github.com/signadot/cli/pull/127/commits/102b36ff3828f93c2cc2a3ba39392fe121c37d73.

**Pull request merged**